pyfr/plugins/fluidforce.py

```diff
@@ -117,10 +140,11 @@ def __call__(self, intg):
 
         # Solution matrices indexed by element type
         solns = dict(zip(intg.system.ele_types, intg.soln))
-        ndims, nvars = self.ndims, self.nvars
+        ndims, nvars, mcomp = self.ndims, self.nvars, self._mcomp
 
-        # Force vector
-        f = np.zeros(2*ndims if self._viscous else ndims)
+        # Force and moment vectors
+        n = 2 if self._viscous else 1
+        fm = np.zeros((ndims + mcomp)*n)
```

Comment:
Can we not keep this as a 2D array to simplify the indexing when we use fm?

Reply:
No that I'm aware since `mcomp` may be different than `ndims`. I tried using lists of arrays but then `mpi4py` may fail because if `mcomp == 0` then one of the arrays has zero length. That's why I tried to use the array splitting or `fp, fv, mp, mv` in the previous commits. It's nicest way to deal with this that I know of. I'm open to suggestions here.

Reply:
So if `fm = np.zeros(ndims + mcomp, 2 if self._viscous else 1)` then:
 - `fm[:ndims, 0] += np.einsum('i...,ij,jik', qwts, p, norms)`
 - `fm[:ndims, 1] += np.einsum('i...,klij,jil', qwts, vis, norms)`
 - `fm[ndims:, 0] += np.einsum(mop, qwts, p, rfpts_c_norms)`
 - `fm[ndims:, 1] += np.einsum('i,jik->k', qwts, rcf)`

The MPI stuff should work unchanged, and all that needs to change is the output routine.  Although here we can simply do `..., *fm.ravel, ` and correct for the different arrangement by changing the headers.